hack/dockerfiles/lint.Dockerfile

```diff
@@ -1,12 +1,11 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION=1.22
-ARG XX_VERSION=1.3.0
-ARG GOLANGCI_LINT_VERSION=1.57.2
-ARG GOPLS_VERSION=v0.20.0
+ARG GO_VERSION=1.23
+ARG XX_VERSION=1.5.0
+ARG GOLANGCI_LINT_VERSION=1.61.0
```

Comment:
Why 61 in 62 is the latest?

Reply:
Was released after my PR got opened :sweat_smile:

Reply:
As you're updating; might as well update this one as well (if it helps; I didn't need to make changes in moby when updating)

Reply:
Done